pkg/api/types.go

```diff
@@ -335,6 +347,9 @@ type ProjectDirectoryImageBuildStepConfiguration struct {
 	// promoted unless explicitly targeted. Use for builds which
 	// are invoked only when testing certain parts of the repo.
 	Optional bool `json:"optional"`
+
+	// IsGitSource defines if the build will be created with a git source.
+	IsGitSource bool `json:"is_git_source"`
```

Comment:
I think the struct embedding would be more appropriate -- factor out the common bits and use them as they need to be used. An API that provides one shape with a switch for behavior is confusing.

Reply:
ProjectDirectoryImageBuildStepConfiguration holds already the fields that we need.

Reply:
it is being used here
https://github.com/openshift/ci-operator/pull/120/files#diff-dd3690bad12236edb7359fb0e2df4ccdR119